masonry/src/widget/zstack.rs

```diff
@@ -0,0 +1,324 @@
+// Copyright 2024 the Xilem Authors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    vello::Scene, widget::WidgetMut, AccessCtx, BoxConstraints, LayoutCtx, PaintCtx, Point,
+    QueryCtx, RegisterCtx, Size, Widget, WidgetId, WidgetPod,
+};
+use accesskit::{NodeBuilder, Role};
+use smallvec::SmallVec;
+use tracing::trace_span;
+
+struct Child {
+    widget: WidgetPod<Box<dyn Widget>>,
+}
+
+#[derive(Default)]
+pub struct ZStack {
+    children: Vec<Child>,
+    alignment: Alignment,
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum Alignment {
+    TopLeading,
+    Top,
+    TopTrailing,
+    Leading,
+    Center,
+    Trailing,
+    BottomLeading,
+    Bottom,
+    BottomTrailing,
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum VerticalAlignment {
+    Top,
+    Center,
+    Bottom,
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum HorizontalAlignment {
```

Comment:
We may want to check whether we can merge or reuse a few of these enums for different container widgets. (no action required here though, I think that should be done in follow-up PRs)